src/Microsoft.AspNetCore.Routing/Matchers/TreeMatcher.cs

```diff
@@ -1,354 +0,0 @@
-            for (var i = 0; i < entries.Count; i++)
-            {
-                var entry = entries[i];
-
-                while (trees.Count <= entry.Order)
-                {
-                    trees.Add(new UrlMatchingTree(entry.Order));
-                }
-
-                var tree = trees[entry.Order];
-                tree.AddEntry(entry);
-            }
-
-            return trees.ToArray();
-        }
-
-        private InboundRouteEntry MapInbound(RouteTemplate template, MatcherEndpoint[] endpoints, int order)
-        {
-            if (template == null)
-            {
-                throw new ArgumentNullException(nameof(template));
-            }
-
-            var entry = new InboundRouteEntry()
-            {
-                Precedence = RoutePrecedence.ComputeInbound(template),
-                RouteTemplate = template,
-                Order = order,
-            };
-
-            // Since all endpoints within a group are expected to have same template and same constraints,
-            // get the first endpoint which has the processor references
-            var endpoint = endpoints[0];
-
-            var matchProcessors = new List<MatchProcessor>();
-            foreach (var matchProcessorReference in endpoint.MatchProcessorReferences)
-            {
-                var matchProcessor = _matchProcessorFactory.Create(matchProcessorReference);
-                matchProcessors.Add(matchProcessor);
-            }
-
-            entry.Tag = new InboundEntryTagData()
-            {
-                Endpoints = endpoints,
-                MatchProcessors = matchProcessors,
-            };
-
-            entry.Defaults = new RouteValueDictionary();
-            foreach (var parameter in entry.RouteTemplate.Parameters)
-            {
-                if (parameter.DefaultValue != null)
-                {
-                    entry.Defaults.Add(parameter.Name, parameter.DefaultValue);
-                }
-            }
-            return entry;
-        }
-
-        private readonly struct Key : IEquatable<Key>
-        {
-            public readonly int Order;
-            public readonly string Template;
-
-            public Key(int order, string routePattern)
-            {
-                Order = order;
-                Template = routePattern;
-            }
-
-            public bool Equals(Key other)
-            {
-                return Order == other.Order && string.Equals(Template, other.Template, StringComparison.OrdinalIgnoreCase);
-            }
-
-            public override bool Equals(object obj)
-            {
-                return obj is Key ? Equals((Key)obj) : false;
-            }
-
-            public override int GetHashCode()
-            {
-                var hash = new HashCodeCombiner();
-                hash.Add(Order);
-                hash.Add(Template, StringComparer.OrdinalIgnoreCase);
-                return hash;
-            }
-        }
-
-        private static class Log
```

Comment:
Is there equivalent logging in the new matcher or do you want to wait until closer to release to flesh out logging?